### DISCUSSION TITLE
Instalace do Home Assistent OS

### DISCUSSION THREAD

**Discussion by Orlando522** [ocervinka/plccoms-mqtt-bridge#10] [Q&A]
Ahoj,
mam Home Assistent OS a PLC CP-1000 jakym zpusobem by se to dalo sparovat za pomoci Tveho PLCComS-MQTT-Bridge ?
Nejak jsem nepobral, jak ten bridge nainstalovat do Home Assistenta.
Moc diky

**Comment by ocervinka** [comment]
Ahoj,
PLCComS-MQTT-Bridge obousmerne convertuje proprietarni protokol PLCComS na univerzalni MQTT protokol. Diky tomu Teco jednotka, napr CP-1000, dokaze komunikovat pres MQTT. Nejedna se o Home Assistant Integraci https://www.home-assistant.io/integrations/ , ale o sluzbu (daemon), ktera stejne jako PLCComS musi bezet samostatne na OS. Home Assistent, nebo jina domaci automatizace se pak pripoji pomoci MQTT https://www.home-assistant.io/integrations/mqtt/

**Reply by pes001** [reply]
Ahoj @ocervinka, resim podobny problem jako je zde popisovan @Orlando522 - propojeni HA s Foxtrotem ridicim topeni a TC Regulus, ze ktereho bych chtel vycitat informace. Neni mi jasne, kde musi PLCComS-MQTT-Bridge bezet, zda na strane PLC, HA serveru nebo zcela samostatne. Nemate nejaky konkretni priklad pouziti teto konverze k integraci s HA? Diky moc Petr

**Reply by Orlando522** [reply]
Ahoj, ja to mam na samostatnem Raspberry Pi, kde bezi PLCComS a pak ten PLCComS-MQTT-Bridge.

**Reply by AcMetelka** [reply]
Ahoj, mě to taky běží oboje na RPi. Ale kde to je, to není důležité, pokud jsou nastavené prostupy, může být bridge klidně na druhé straně světa. Jednodušší samo na stejné LAN. Potřeba jen v tom Bridge nastavit na obě strany správnou IP a port. 
Pokud se ptáš na propojení/posloupnost, tak tedy PLC > PLCComS > Bridge > HA.

**Reply by pes001** [reply]
Mozna je to nesmysl, a omlouvam se za neznalost, ale na stejnem RPi kde bezi HA to asi bezet nemuze, ze? A PLCComS nebezi uz na Foxtrotu, ten je treba mit taky samostatne? Zatim s Foxtrotem komunikuji jen pres web rozhrani, resp. taky Teco route. Diky P.

**Reply by AcMetelka** [reply]
Novější Foxtroty myslím mají PLCComS zabudovaný, ale já mám starý Foxtrot verze 1 a tam musím mít PLCCOmS někde mimo. Dá se např. mít na domácím routeru, pokud dovolí instalaci a bude sedět architektura procesoru s verzemi, co TECO nabízí. Aby PLCComS, Bridge i HA běženo na jednom HW, není problém, já to mám vše na RPi. Bridge i HA mám v dockeru.

**Reply by ocervinka** [reply]
1. PLC (Inels CU2-01M, Tecomat Foxtrot CP-1xxx, CP-2xxx)
2. plccoms
3. plccoms-mqtt-bridge
4. MQTT broker
5. Home Assistent (s MQTT integraci https://www.home-assistant.io/integrations/mqtt/)

* 2, 3, 4 a 5 muzou klidne bezet na jedom HW (treba RPi). 
* 2 (plccoms) bezi v novych jednotkach Foxtrot, takze se o nej mozna nemusis vubec starat. Nevim ale napr. zda je treba explicitne urcit promemne pro export nebo se exportuje vse automaticky. (Ja mam stale jen Inels CU2-01M, takze plccoms konfiguruji a provozuji samostane).
* 4 (MQTT broker) Je soucasti nekterych distribuci Home Assistent. (Ja ho provozuji samostatne - Mosquitto MQTT broker)
* Jestli to HW a OS dovoli, tak je prakticke vsechny processy spoustet jako Docker kontejnery a konfigurovat a udrzovat v kondici centralne napr. pomoci Docker Compose nebo Kubernetes.

**Reply by pes001** [reply]
Ahoj, po case jsem se dostal k pokracovani vytazeni dat z Foxtrotu do HA ale nemohu to rozchodit. Bezi mi plccoms na dockeru v Synology, MQTT broker Mosquito mi bezi primo v HA (RPI3B+) ale nejde mi spustit plccoms-mqtt-bridge - poustim take v dockeru na Synology. V config.yaml mam toto: 
![image](https://user-images.githubusercontent.com/73618038/187206871-69d2c7f4-fcea-4fb6-bbeb-382dadfa168d.png)
Pokud tam necham username a password tak to nezna:
_Exception in thread "main" com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "username" (class ocervinka.plcmqttbridge.mqtt.MqttConfig), not marked as ignorable (4 known properties: "host", "port", "scheme", "clientId"])_
a pokud usera a password vymazu tak to konci na neautorizovanem pristupu (k mqtt brokeru predpokladam).
_Exception in thread "main" Not authorized to connect (5)_ 
Umeli byste mi pomoci, mozna delam neco uplne blbe? Diky moc P

**Reply by ocervinka** [reply]
Ahoj,

Nepouzivas starsi verzi?

Parametry `username` a `password` byly pridany pozdeji (v0.2 https://github.com/ocervinka/plccoms-mqtt-bridge/releases/tag/v0.2), ale uz jsou nejakou dobu v master: https://github.com/ocervinka/plccoms-mqtt-bridge/blob/6f5d19de51f0e65b7dd6cb7d6ab227786aeb2ba2/src/main/java/ocervinka/plcmqttbridge/mqtt/MqttConfig.java#L28

Presne tak, kdyz `username` a `password` v konfiguracnim souboru chybi (nebo pouzivas starsi verzi), `plccoms-mqtt-bridge` se bude pripojovat jako neautorizovany a je brokerem odmitnut (za predpokladu ze broker nema povoleny neautorizovany pristup).

**Reply by ocervinka** [reply]
Pouzivas docker image z https://hub.docker.com/r/ocervinka/plccoms-mqtt-bridge/tags? Pri kazdym buildu by tam Github Action mela nahrat novy image.

Ale ted vidim ze `latest` je dva roky stary a `v0.2` tam uplne chybi. Muzes pouzit `v0.1-8-g6f5d19d` coz je posledni build na master (jen neni jeste tagovany jako `v0.2`.

**Reply by pes001** [reply]
Diky, dnes nebo o vikendu to vyzkousim. Pouzival jsem "latest". Jeste tam vidim verzi v0.1-9-g2a4b917.

**Reply by pes001** [reply]
Tak s verzi v v0.1-9-g2a4b917 mi to bezi ale zase se mi to nechce napojit na PLCCOMs.
![image](https://user-images.githubusercontent.com/73618038/187907168-f5f07872-5834-48bf-99cc-08db9e343331.png)
Chybi mi tam nejaka konfigurace? Mam tam jen to co je ve vyse uvedenem config.yaml, zkousel jsem i misto IP localhost ale to stejne (plccoms bezi na stejnem dockeru v synology jako PLCComS-MQTT-Bridge). Port plccoms mam nastaven takto: 
![image](https://user-images.githubusercontent.com/73618038/187908191-87181aee-c574-4ef1-ae8c-648ff567ae87.png)
(v konfiguraku je auto/5010).
Diky moc

**Comment by ocervinka** [comment]
```
plccoms:
  host: localhost
  port: 5010
```

Je spravny format konfigurace. `localhost`, znamena ze `plccoms-mqtt-bridge` se bude pripojovat k processu plccoms uvnitr sveho kontejneru na port 5010. To je v poradku pokud oba processy opravdu bezi ve stejnem Docker kontejneru. Obvykla praxe je ze bezi kazdy ve svek kontejneru. 

```
plccoms:
  host: adresa/jmeno docker host (pravdepodobne adresa Synology v LAN)
  port: exposed port plccoms kontejneru na docker host, napr. 49xxx 
```

Je ale potreba vedet, jak se na Synolgy exposed port definuji.

**Reply by pes001** [reply]
Zda se, ze se mi to podarilo spustit, diky moc! :) Nenapadlo me, ze ten port je v configu je ten lokalni 49xxx. 
![image](https://user-images.githubusercontent.com/73618038/188514909-d043f922-8cce-429e-bf15-21e8baad3edd.png)
Tak jeste najit ty spravne promenne a namapovat je, je jich tam pres 2000 a potrebuju cca 20-30 ;)

**Reply by ocervinka** [reply]
Jestli mas CU2-01M naprogramovanou pres FoxTool (nebo jeste stary IDM SW od Inels), tak by melo byt pouzitelne mapovani z [konfigu tady](https://github.com/ocervinka/plccoms-mqtt-bridge/blob/master/deploy/docker/config.yaml). Pro program z Mosaic si ho ale budes asi muset napsat od zacatku.

Drzim palce!
